### ISSUE TITLE
Bug: The argument 'path' must be a string or Uint8Array without null bytes

### ISSUE THREAD

**Issue by tricki** [ionic-team/stencil#2520]
**Stencil version:**

```
 @stencil/core@1.15.0-4
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**
Building causes the following type of error multiple times and hangs the build process (i.e. it never finishes).

```
[ ERROR ]  unhandledRejection: TypeError [ERR_INVALID_ARG_VALUE]: The argument 'path' must be a string or Uint8Array
           without null bytes. Received
           'C:/tests/stencil-import-path-bug/\u0000C:/tests/stencil-import-path-bug/node_modules/dragula'
           at Object.realpath (fs.js:1687:3) at
           C:\tests\stencil-import-path-bug\node_modules\@stencil\core\cli\index.js:5179:29 at new Promise
           (<anonymous>) at Object.realpath
           (C:\tests\stencil-import-path-bug\node_modules\@stencil\core\cli\index.js:5178:20) at
           Object.fs.realpath
           (C:\tests\stencil-import-path-bug\node_modules\@stencil\core\compiler\stencil.js:3001:12) at
           maybeUnwrapSymlink
           (C:\tests\stencil-import-path-bug\node_modules\@stencil\core\compiler\stencil.js:13274:8) at
           resolve
           (C:\tests\stencil-import-path-bug\node_modules\@stencil\core\compiler\stencil.js:13312:3) at
           C:\tests\stencil-import-path-bug\node_modules\@stencil\core\compiler\stencil.js:2915:10 at new
           Promise (<anonymous>) at
           C:\tests\stencil-import-path-bug\node_modules\@stencil\core\compiler\stencil.js:2906:12

```

**Expected behavior:**
Build is successful.

**Steps to reproduce:**
I've managed to reproduce the bug by importing [`dragula`](https://github.com/bevacqua/dragula):
1. clone the [Example repository](https://github.com/tricki/stencil-import-path-bug)
2. run `npm run build`

My steps:
1. Create new component starter
2. Install `dragula`
3. Use `dragula` in `my-component.tsx`
4. run `npm run build`

**Related code:**
While trying to debug the source of this error I got as far as [`resolveId`](https://github.com/ionic-team/stencil/blob/master/src/compiler/bundle/plugin-helper.ts#L8) which receives two parameters:
- `importee`: `"C:\tests\stencil-import-path-bug\node_modules\dragula\classes.js"`
- `importer`: `" C:\xampp\htdocs\tests\stencil-import-path-bug\node_modules\dragula\classes.js?commonjs-proxy"` 

Note: the first character of `importer` is actually a null byte (`'\u0000'`) which I can't paste in the GitHub editor.

**Other information:**

The same (or a similar) bug was reported [in the Stencil Slack](https://stencil-worldwide.slack.com/archives/C79EANFL7/p1591805736325100) after an update from 1.7.1 to 1.14.0)

**Comment by tricki** [comment]
I've managed to trace this as far back as Rollup's [`fetchAllDependencies` method](https://github.com/rollup/rollup/blob/5028da3a27b703a2ced9b02645ff7aea136b0aad/src/Bundle.js#L234) where I'm seeing multiple entries in `module.sources` which start with a null byte:

```
[
  "\u0000commonjsHelpers.js", 
  "contra/emitter", 
  "crossvent", 
  "./classes", 
  "\u0000contra/emitter?commonjs-proxy", 
  "\u0000crossvent?commonjs-proxy", 
  "\u0000./classes?commonjs-proxy"
]
```

Since the error only appears for the last entry and not for the previous ones which also start with a null byte I suspect that this null byte is possibly a Rollup feature but I have no experience working with Rollup.

Any help or pointers on debugging this are appreciated.

**Comment by peterpeterparker** [comment]
For traceability: this seems like the same error as #2461 and #2494

**Comment by peterpeterparker** [comment]
Found [this](https://github.com/rollup/rollup-plugin-commonjs/issues/397#issuecomment-507677825) about \0

> If your plugin uses 'virtual modules' (e.g. for helper functions), prefix the module ID with \0. This prevents other plugins from trying to process it.

**Comment by peterpeterparker** [comment]
Found [this](https://github.com/rollup/rollup-plugin-commonjs/issues/397#issuecomment-507677825) about \0

> If your plugin uses 'virtual modules' (e.g. for helper functions), prefix the module ID with \0. This prevents other plugins from trying to process it.

**Comment by patrickhaug** [comment]
I got the same error with a graphql lib. there is a "/\x00/" injected in path .. but only when upgrading to latest stencil version

**Comment by peterpeterparker** [comment]
Fixed to me with v1.15. This issue can be closed I think.
